docs/design/StratisSoftwareDesign.lyx

```diff
@@ -6933,5 +6933,423 @@ Based on looking at the existing available building blocks, the best option
+ attacker being able to determine exactly how much of the data on the device
+ is encrypted, the device should be wiped with random data prior to providing
+ it to a Stratis pool.
+ If a user does not want data that was previously stored on the device (encrypte
+d or unencrypted) to be visible, the disk should be zeroed or wiped with
+ random data.
+\end_layout
+
+\begin_layout Subsection
+Encrypted device discovery
+\end_layout
+
+\begin_layout Standard
+To find encrypted Stratis devices, stratisd searches for LUKS2 volumes present
+ on the system and validates that the LUKS2 metadata on the device is well
+ formed LUKS2 metadata laid down by Stratis.
+ Once this has been verified, Stratis stores a record of this device and
+ waits for the user to activate the LUKS2 volume which can be initiated
+ by a D-Bus API method.
+ The devices will be unlocked with the corresponding passphrase in the kernel
+ keyring as specified by the key description in the LUKS2 keyring token.
+ Stratis will then search all resulting unencrypted logical devices for
+ the Stratis metadata and will construct pools accordingly.
+\end_layout
+
+\begin_layout Subsection
+Encrypted device activation
+\end_layout
+
+\begin_layout Standard
+Devices are activated with the activation name in the Stratis token.
+ See 
+\begin_inset CommandInset ref
+LatexCommand ref
+reference "subsec:Stratis-token"
+plural "false"
+caps "false"
+noprefix "false"
+
+\end_inset
+
+ for an explanation of the Stratis token.
+ libcryptsetup handles device activation and should result device with a
+ path of 
+\begin_inset listings
+inline false
+status collapsed
+
+\begin_layout Plain Layout
+
+/dev/mapper/<activation_name>
+\end_layout
+
+\end_inset
+
+ where the unencrypted data can be accessed.
+\end_layout
+
+\begin_layout Subsection
+Encrypted device destruction
+\end_layout
+
+\begin_layout Standard
+Destroying an encrypted Stratis device does not wipe the entire device due
+ to the time that this would take increasing linearly with regard to disk
+ size.
+ Instead, Stratis uses libcryptsetup to destroy all of the keyslots, wiping
+ all of the encrypted MEK (media encryption key) data in each keyslot.
+ It then obtains the size of the LUKS2 metadata on this device and does
+ an additional wipe of the remaining LUKS2 metadata.
+ This provides the following device properties on pool destruction:
+\end_layout
+
+\begin_layout Itemize
+The MEK data is destroyed so the data stored on the device will be unrecoverable.
+\end_layout
+
+\begin_layout Itemize
+If this device is used when creating another pool later, Stratis will be
+ able to reinitialize it and will consider this device unowned due to the
+ wipe of the LUKS2 metadata.
+\end_layout
+
+\begin_layout Itemize
+The encrypted data will be left on the device but will be inaccessible due
+ to the destruction of the key.
+\end_layout
+
+\begin_layout Standard
+If the user is concerned about leaving the encrypted data on the device
```

Comment:
Since this is something that Stratis does not do, you could say something like:

```
Additional security may be obtained by further overwriting the encrypted data using external tools.
```

rather than being too user-directed. Otherwise, you have to address the question of why the user might be concerned and that's a long tale.